java/arcs/android/storage/database/AndroidSqliteDatabaseManager.kt

```diff
@@ -12,27 +12,73 @@
 package arcs.android.storage.database
 
 import android.content.Context
+import android.content.ContextWrapper
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.OnLifecycleEvent
 import arcs.core.storage.database.Database
 import arcs.core.storage.database.DatabaseIdentifier
 import arcs.core.storage.database.DatabaseManager
 import arcs.core.storage.database.DatabasePerformanceStatistics.Snapshot
+import arcs.core.util.Log
 import arcs.core.util.guardedBy
 import kotlinx.coroutines.Job
 import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
 import kotlinx.coroutines.sync.Mutex
 import kotlinx.coroutines.sync.withLock
 
 /**
  * [DatabaseManager] implementation which constructs [DatabaseImpl] instances for use on Android
  * with SQLite.
  */
-class AndroidSqliteDatabaseManager(context: Context) : DatabaseManager {
+class AndroidSqliteDatabaseManager(
+    context: Context,
+    lifecycleParam: Lifecycle? = null
+) : DatabaseManager, LifecycleObserver {
     private val context = context.applicationContext
+    private val lifecycle = lifecycleParam?.let { it } ?: getLifecycle()
```

Comment:
This can just be `lifecycleParam ?: getLifecycle()` can't it?

Reply:
Yep!

Reply:
its a leftover from when lifecycleParam was named 'lifecycle' and my IDE was getting confused by scopes (param vs field) but somehow .let fixed the squiggly. :(